### ISSUE TITLE
Unwanted ANSI Control Chars when knitting to HTML in RStudio

### ISSUE THREAD

**Issue by thothal** [r-lib/pillar#318]
## Disclaimer

I posted an [SO question](https://stackoverflow.com/questions/67332147/avoid-printing-console-control-chars-when-knitting-via-rstudio) , but I think it is more of a bug, hence a double post here.

## Issue

When using `knitr::knit2html` to display some html, he output depends on the GUI. While everything renders fine (i.e. no control characters int eh output), when issued form the R console, runnning the same code from RStudio results in control charatcers to be displayed in the HTML. We can partly switch off certain control characters (notably the once controlled via `pillar.subtle` but not all of them (the italic type labels). to render the output w/o control charatcers I have to manully remove them via `crayon::strip_style`. Tryin to restrict the number of colors to `1` via `cli.num_colors` had no effect.

## Reprex

```r
library(shiny)
library(knitr)
library(tibble)
library(glue)

ui <- fluidPage(checkboxInput("use_subtle", "pilar.subtle", value = TRUE), 
                checkboxInput("strip", "Use strip_style", value = FALSE),
                uiOutput("content"))

server <- function(input, output) {
   n <- new.env()
   output$content <- renderUI({
      subtle <- input$use_subtle
      chunk <- glue("```{{r, R.options = list(pillar.subtle = {subtle}, cli.num_colors=1)}}
                        as_tibble(mtcars)
                     ```")
      res <- HTML(knit2html(text = chunk, fragment.only = TRUE, quiet = TRUE, envir = n))
      if (input$strip) {
         res <- crayon::strip_style(res)
      }
      res
   })
}

shinyApp(ui, server)
```

### R  Studio
![ControlChars](https://user-images.githubusercontent.com/30824940/116692684-0ef85180-a9bd-11eb-8fd1-be40e10638aa.gif)

### R Console
![ControlChars_R](https://user-images.githubusercontent.com/30824940/116692716-17508c80-a9bd-11eb-9fa9-7c9b1bdb5921.gif)

## Session Info

```
R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252   
[3] LC_MONETARY=German_Germany.1252 LC_NUMERIC=C                   
[5] LC_TIME=German_Germany.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] pillar_1.6.0 tibble_3.1.1

loaded via a namespace (and not attached):
 [1] compiler_3.6.3  magrittr_2.0.1  ellipsis_0.3.1  crayon_1.4.1   
 [5] utf8_1.2.1      fansi_0.4.2     vctrs_0.3.7     lifecycle_1.0.0
 [9] pkgconfig_2.0.3 rlang_0.4.10   
```

**Comment by krlmlr** [comment]
Thanks. I can't replicate. What version of RStudio are you using?

**Comment by thothal** [comment]
Holy cow. I tried it again today and guess what, it worked. No idea what went wrong in the first place, it's just good that I made a small gif, otherwise I'd believe that I am crazy. 

Any ideas which extraterrestrial forces may have lead to the captured behavior? Sunspots? Moody Monday computers? From what I could judge, I did not update any package in the meantime (and I figured that I should have included the `sessionInfo()` after the reprex and not just with `tibble/pillar` loaded.
